### PR TITLE
Add status view

### PR DESCRIPTION
**Pull request by frederoni**
The rerouting UI was removed in #464 
This PR adds a new reusable status view that also acts as a rerouting UI.

@bsudekum @ericrwolfe 👀 

_The rerouting ui is presented manually in this gif_

### CODE REVIEW

File: MapboxNavigation/RouteMapViewController.swift [hunk 1]
```diff
@@ -204,6 +220,11 @@ class RouteMapViewController: UIViewController {
             wayNameView.isHidden = true
         }
     }
+    
+    func willReroute(notification: NSNotification) {
+        let title = NSLocalizedString("REROUTING", bundle: .mapboxNavigation, value: "Rerouting…", comment: "Indicates that rerouting is in progress")
```
Comment (bsudekum) [inline MapboxNavigation/RouteMapViewController.swift]: Note, we'll need to update the translation files.

File: MapboxNavigation/Style.swift [hunk 2]
```diff
@@ -512,3 +514,51 @@ class ManeuverContainerView: UIView {
...
+    @IBOutlet weak var activityIndicatorView: UIActivityIndicatorView!
+    @IBOutlet weak var textLabel: UILabel!
+    @IBOutlet weak var topConstraint: NSLayoutConstraint!
+    
+    func show(_ title: String, showSpinner: Bool, duration: TimeInterval) {
+        textLabel.text = title
+        activityIndicatorView.isHidden = !showSpinner
+        activityIndicatorView.startAnimating()
+        
+        show()
```
Comment (bsudekum) [inline MapboxNavigation/Style.swift]: Maybe call these `show()` function something a little different? hard to know what they each do.

File: MapboxNavigation/RouteMapViewController.swift (hunk 1 above)
Comment (frederoni) [inline MapboxNavigation/RouteMapViewController.swift]: Included the `extract_localizable.sh` script. Seems to work great.

File: MapboxNavigation/Style.swift (hunk 2 above)
Comment (frederoni) [inline MapboxNavigation/Style.swift]: They're fileprivate so shouldn't affect too much, but if you have a better name 😊 
How about `updateConstraints(showOrHide: Bool)`? 🤔

Comment (frederoni) [inline MapboxNavigation/Style.swift]: Fixed in https://github.com/mapbox/mapbox-navigation-ios/pull/468/commits/b6a51bb351e8d7c5f3848acfe27f4122456934cd

**Pull request merged**